src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/CovariantEqualsCheck.java

```diff
@@ -144,8 +144,7 @@ public void visitToken(DetailAST ast) {
             DetailAST child = objBlock.getFirstChild();
             boolean hasEqualsObject = false;
             while (child != null) {
-                if (child.getType() == TokenTypes.METHOD_DEF
-                        && CheckUtil.isEqualsMethod(child)) {
```

Comment:
`CheckUtil.isEqualsMethod` already checks for METHOD_DEF.

Reply:
https://github.com/checkstyle/checkstyle/blob/bfc6adf4c43afa24a6f2c67724500a93b8527124/src/main/java/com/puppycrawl/tools/checkstyle/utils/CheckUtil.java#L99